tests/multi_syscall_test.rs

```diff
@@ -0,0 +1,288 @@
+        let call_info = test_syscall(
+            "test_send_message_to_l1",
+            address.clone(),
+            new_call_data,
+            caller_address.clone(),
+            entry_point_type,
+            class_hash,
+            &mut state,
+        );
+        assert_eq!(call_info.retdata, vec![])
+    }
+
+    // Block for read write
+    {
+        let call_info = test_syscall(
+            "read",
+            address.clone(),
+            calldata.clone(),
+            caller_address.clone(),
+            entry_point_type,
+            class_hash,
+            &mut state,
+        );
+        assert_eq!(call_info.events, vec![])
+    }
+
+    // Block for emit
+    {
+        let call_info = test_syscall(
+            "trigger_events",
+            address.clone(),
+            calldata.clone(),
+            caller_address.clone(),
+            entry_point_type,
+            class_hash,
+            &mut state,
+        );
+        assert_eq!(
+            call_info.events,
+            vec![
+                OrderedEvent {
+                    order: 0,
+                    keys: vec![Felt252::from_str_radix(
+                        "1533133552972353850845856330693290141476612241335297758062928121906575244541",
+                        10
+                    )
+                    .unwrap()],
+                    data: vec![1.into()]
+                },
+                OrderedEvent {
+                    order: 1,
+                    keys: vec![Felt252::from_str_radix(
+                        "1533133552972353850845856330693290141476612241335297758062928121906575244541",
+                        10
+                    )
+                    .unwrap()],
+                    data: vec![2.into()]
+                },
+                OrderedEvent {
+                    order: 2,
+                    keys: vec![Felt252::from_str_radix(
+                        "1533133552972353850845856330693290141476612241335297758062928121906575244541",
+                        10
+                    )
+                    .unwrap()],
+                    data: vec![3.into()]
+                }
+            ]
+        )
+    }
+
+    // Block for deploy syscall
+    {
+        // data to deploy
+        let test_class_hash: ClassHash = [2; 32];
+        let test_data = include_bytes!("../starknet_programs/cairo1/contract_a.casm");
+        let test_contract_class: CasmContractClass = serde_json::from_slice(test_data).unwrap();
+
+        // Create the deploy contract class
+        contract_class_cache.insert(class_hash, contract_class);
+        contract_class_cache.insert(test_class_hash, test_contract_class);
+        let call_info = test_syscall(
+            "deploy_test",
+            address,
+            calldata,
+            caller_address,
+            entry_point_type,
+            class_hash,
+            &mut state,
+        );
+        assert_eq!(call_info.events, vec![])
```

Comment:
Do you think you can test calling the deployed contract so you check it was deployed successfully?